internal/controllers/validators/validators_test.go

```diff
@@ -1,63 +1,94 @@
+	{"1.x", "with lower-case y-stream", true},
+	{"1.X", "with upper-case Y-stream", true},
+	{"1.*", "with asterisk y-stream", true},
+	{"1.2.x", "with lower-case z-stream", true},
+	{"1.2.X", "with upper-case Z-stream", true},
+	{"1.2.*", "with asterisk z-stream", true},
+	{">=1.2.3 <2.3.4", "multiple operators space-separated", true},
+	{">=1.2.3,<2.3.4", "multiple operators comma-separated", true},
+	{">=1.2.3, <2.3.4", "multiple operators comma-and-space-separated", true},
+	{"<1.2.3||>2.3.4", "multiple operators OR-separated", true},
+	{"<1.2.3|| >2.3.4", "multiple operarors OR-and-space-separated", true},
+	{"<1.2.3 ||>2.3.4", "multiple operators space-and-OR-separated", true},
+	{"<1.2.3 || >2.3.4", "multiple operators space-and-OR-and-space-separated", true},
+	{">1.0.0,<1.2.3 || >2.1.0", "multiple operators with comma and OR separation", true},
+	{"<1.2.3-abc >2.3.4-def", "multiple operators with pre-release data", true},
+	{"<1.2.3-abc+def >2.3.4-ghi+jkl", "multiple operators with pre-release and metadata", true},
+	// list of invalid semvers
+	{"invalid-semver", "invalid characters", false},
+	{"1.2.3.4", "too many components", false},
+	{"1.2.3-beta!", "invalid character in pre-release", false},
+	{"1.2.3.alpha", "invalid pre-release/4th component", false},
+	{"1..2.3", "extra dot", false},
+	{"1.2.3-pre+bad_metadata", "invalid metadata", false},
+	{"1.2.-3", "negative component", false},
+	{".1.2.3", "leading dot", false},
+	{"<<1.2.3", "invalid << operator", false},
+	{">>1.2.3", "invalid >> operator", false},
+	{">~1.2.3", "invalid >~ operator", false},
+	{"==1.2.3", "invalid == operator, valid for blang", false},
+	{"=!1.2.3", "invalid =! operator", false},
+	{"!1.2.3", "invalid ! operator, valid for blang", false},
+	{"1.Y", "invalid y-stream wildcard", false},
+	{">1.2.3 && <2.3.4", "invalid AND separator", false},
+	{">1.2.3;<2.3.4", "invalid semicolon separator", false},
+	// Invalid semvers that get past simple validation - THESE NEED TO BE TESTED SEPARATELY
+	{"1.02.3", "leading zero in y-stream - FALSE POSITIVE", true},
+	{"1.2.03", "leading zero in z-stream - FALSE POSITIVE", true},
+	{"1.2.3 - 2.3.4", "unsupported hyphen (range) operator - FALSE POSITIVE", true},
+}
 
-		It("should not return an error for valid SemVer with pre-release", func() {
-			operator := &v1alpha1.Operator{
-				Spec: v1alpha1.OperatorSpec{
-					Version: "1.2.3-alpha-beta",
-				},
-			}
-			err := validators.ValidateOperatorSpec(operator)
-			Expect(err).NotTo(HaveOccurred())
-		})
-	})
-})
+func TestValidateOperatorSpecSemVer(t *testing.T) {
+	for _, d := range sm {
+		t.Logf("Testing %s: %q", d.comment, d.data)
+		operator := &v1alpha1.Operator{
+			Spec: v1alpha1.OperatorSpec{
+				Version: d.data,
+			},
+		}
+		if d.result {
+			require.NoError(t, validators.ValidateOperatorSpec(operator))
+		} else {
+			require.EqualError(t, validators.ValidateOperatorSpec(operator), fmt.Sprintf("invalid .spec.version: improper constraint: %s", d.data))
```

Comment:
Seems like an overly brittle assertion on the specific format of the human-readable validation error - do we care about the specific contents here? Why? Will anyone ever programmatically look into this error message?

Reply:
Can't answer the "why", but trying to keep the "what" as similar as possible. Although, AFAICT, all the errors are basically the same, so little point in checking the text.